pkg/apis/core/helper/qos/qos.go

```diff
@@ -30,6 +30,15 @@ func isSupportedQoSComputeResource(name core.ResourceName) bool {
 	return supportedQoSComputeResources.Has(string(name))
 }
 
+// PodQOSClass returns the QoS class of a pod persisted in the PodStatus.
+// If QOSClass is empty, it returns value of GetPodQOS() which computes pod's QoS class.
+func PodQOSClass(pod *core.Pod) core.PodQOSClass {
+	if pod.Status.QOSClass != "" {
+		return pod.Status.QOSClass
+	}
+	return GetPodQOS(pod)
```

Comment:
Maybe note that this is much more expensive?

Reply:
Or even rename to GetPodQOSExpensive() or something?

Reply:
Or name this  `GetPodQOS()` and move the old, expensive one to `getPodQOSExpensive()`

Reply:
That's a great suggestion, fewer changes! I'll rename the current expensive one as ComputePodQOS() or DeterminePodQOS(), I'll update & retest over the weekend.

Reply:
@thockin My last change has addressed this. PTAL, ty